src/classes/CON_PrimaryAffls_TDTM.cls

```diff
@@ -51,151 +51,200 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
+    * @description Returns if Contact Deletion Prevention custom setting is not enabled or
+    * Passes a map of contacts By Id to SRVC_Contact_PreventDeletion and returns a List of Contacts that have
+    * been prevented for deletion
+    * @param oldSObjectList list of sObject records that are being deleted
+    * @return List<Contact>
+    ********************************************************************************************************/
+    @TestVisible
+    private void handleAfterUpdate(List<SObject> oldSObjectList, List<SObject> newSObjectList) {
+
+        if (this.isUpdateRecursion() == true) {
+            return;
+        }
         
-        //Initially we get all possible affiliations from those contacts.
-        List<Affiliation__c> afflsAll = [select Contact__c, Account__c, Primary__c from Affiliation__c where Contact__c in :changedConToAccsMap.keySet()];
-        UTIL_Debug.debug('****All possible affls to make non-primary: ' + JSON.serializePretty(afflsAll));
         
-        //Then we refine the search - changedConToAccsMap should contain only the Acc IDs of those key Affl lookups that have been modified.
-        for (Affiliation__c affl : afflsAll) {
-            List<ID> contactAccIDs = changedConToAccsMap.get(affl.Contact__c);
-            for (ID changedAccID : contactAccIDs) {
-                if (affl.Account__c == changedAccID) {
-                    affl.Primary__c = false;
-                    UTIL_Debug.debug('****Making affl ' + affl.ID + ' non-primary.');
-                    afflsToMakeNonPrimary.add(affl);
-                }
+        setUpdateRecursion(); 
+
+        Map<Id, Contact> oldContactById = new Map<Id, Contact>((List<Contact>)oldSObjectList);
+        Map<Id, Contact> newContactById = new Map<Id, Contact>((List<Contact>)newSObjectList);
+
+        List<MODL_PrimaryAfflMapper> primaryAffiliationModels = this.locateAffiliationSettingMapper().getModelsFromSettings();
+
+        SRVC_Contact_PrimaryAffiliations.AffiliationsToInsertAndUpdate affiliationsToInsertAndUpdate = this.locateContactAffiliationService().getAffiliationsToCreateOrUpdate(newContactById, oldContactById, primaryAffiliationModels);
+        
+        DmlWrapper dmlWrapper = new DmlWrapper();
+
+        if (AffiliationsToInsertAndUpdate.getAffiliationListToUpdateClone().isEmpty() == false) {
+            this.setAffiliationUpdateRecursion();
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)(AffiliationsToInsertAndUpdate.getAffiliationListToUpdateClone()));
+        }
+
+        if (AffiliationsToInsertAndUpdate.getAffiliationListToinsertClone().isEmpty() == false) {
+            if (this.isAffiliationUpdateRecursion() == false) {
+                this.setAffiliationUpdateRecursion();
             }
+
+            dmlWrapper.objectsToInsert.addAll((List<SObject>)(AffiliationsToInsertAndUpdate.getAffiliationListToInsertClone()));
```

Comment:
Should these be added to objectsToUpdate?

Reply:
This is the list of affiliations to insert @xinnanlisf when we update the Primary field on Contact.